packages/network/src/createTxQueue.ts

```diff
@@ -190,7 +190,9 @@ export function createTxQueue<C extends Contracts>(
         gasLimit = await txRequest.estimateGas();
       } catch (e) {
         console.error("[TXQueue] GAS ESTIMATION ERROR", e);
-        return txRequest.cancel();
+        throw e;
+      } finally {
+        txRequest.cancel();
```

Comment:
this cancels every tx

Reply:
ahh I see, so it makes sure that if one gas estimate fails, the whole queue is canceled?

Reply:
https://github.com/latticexyz/mud/blob/f18c39831f5d5ab0186529eb2cfaee4991570e6f/packages/network/src/createTxQueue.ts#L155

I see now that this just throws a generic error. Maybe I should rip out `txRequest.cancel` and just let the real error bubble up.

Reply:
> this cancels every tx

I see what you mean now 🙈